modules/shared_data/src/main/scala/org/elpaca_metagraph/shared_data/types/YouTube.scala

```diff
@@ -0,0 +1,77 @@
+package org.elpaca_metagraph.shared_data.types
+
+import derevo.circe.magnolia.{decoder, encoder}
+import derevo.derive
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.balance.Amount
+import org.tessellation.schema.epoch.EpochProgress
+
+import java.time.{Instant, LocalDateTime}
+import scala.collection.immutable.ListMap
+
+object YouTube {
+  @derive(encoder, decoder)
+  case class YouTubeRewardInfo(dailyEpochProgress: EpochProgress,
+                               epochProgressToReward: EpochProgress,
+                               amountToReward: Amount,
+                               publishDate: LocalDateTime)
+
+  @derive(encoder, decoder)
+  case class YouTubeDataSourceAddress(videoRewards: ListMap[String, YouTubeRewardInfo] = ListMap.empty,
+                                      rewardsReceivedToday: Long = 0)
+
+  object LatticeClient {
+    @derive(encoder, decoder)
+    case class YouTubeAccount(channelId: String)
+
+    @derive(encoder, decoder)
+    case class LatticeUser(id: String,
+                           primaryDagAddress: Option[Address],
+                           youtube: Option[YouTubeAccount])
+
+    @derive(encoder, decoder)
+    case class LatticeUserMeta(total: Long,
+                               limit: Long,
+                               offset: Long)
+
+    @derive(encoder, decoder)
+    case class LatticeUsersApiResponse(data: List[LatticeUser],
```

Comment:
I saw your comment for future changes, we definitely need to unify this to be used for all data sources that require lattice users, for now, it's good to leave it as it is 👍🏼